src/components/Navigation/components/Item/Item.tsx

```diff
@@ -40,7 +40,7 @@ interface SecondaryAction {
 export interface Props extends ItemURLDetails {
   icon?: IconProps['source'];
   iconBody?: string;
-  badge?: string | null;
+  badge?: string | React.ReactElement<BadgeProps> | null;
```

Comment:
`BadgeProps` doesn't help here, really. The component you have in the Web PR doesn't conform to that prop shape, unless all Badge's props are optional, but that's OK. You don't depend on the props for anything, so I would just make this prop a `React.ReactNode` instead and accept anything

Reply:
Right, I overlooked that those interfaces don't match. Will update.